packages/react/core/src/features/useConnect.ts

```diff
@@ -0,0 +1,50 @@
+import type { StandardConnectFeature } from '@wallet-standard/features';
+import { StandardConnect } from '@wallet-standard/features';
+import { useCallback } from 'react';
+
+import type { ReactWallet } from '../ReactWallet.js';
+import type { ReactWalletAccount } from '../ReactWalletAccount.js';
+import { getOrCreateReactWalletAccountForStandardWalletAccount_INTERNAL_ONLY_NOT_FOR_EXPORT } from '../WalletAccountRegistry_INTERNAL_ONLY_NOT_FOR_EXPORT.js';
+import { getWalletForHandle_DO_NOT_USE_OR_YOU_WILL_BE_FIRED } from '../WalletHandleRegistry_INTERNAL_ONLY_NOT_FOR_EXPORT.js';
+import { useWeakRef } from '../useWeakRef.js';
+import { useWalletFeature } from './useWalletFeature.js';
+
+/**
+ * Returns a function you can call to ask the wallet to authorize the current domain to use new
+ * accounts.
+ */
+export function useConnect<TWallet extends ReactWallet>(
+    reactWallet: TWallet
+): [isConnecting: boolean, connect: () => Promise<readonly ReactWalletAccount[]>] {
+    const connectFeature = useWalletFeature(
+        reactWallet,
+        StandardConnect
+    ) as StandardConnectFeature[typeof StandardConnect];
+    const wallet = getWalletForHandle_DO_NOT_USE_OR_YOU_WILL_BE_FIRED(reactWallet);
+    const connectionPromise = useWeakRef<Promise<readonly ReactWalletAccount[]>>(wallet);
+    const connect = useCallback(
+        async function connect() {
+            if (connectionPromise.current) {
+                return connectionPromise.current;
+            }
+            const newConnectionPromise = connectFeature
+                .connect({ silent: false })
```

Comment:
Ah, so yes – this does make it impossible to call `{ silent: false }` and that was on purpose.

In my opinion, the _only_ thing this method should do is to seek additional authorizations for new accounts, all the time, every time it's called.

Otherwise, wallets should populate the `accounts` property with the currently-authorized-accounts, as per the spec.

They _definitely_ don't do that today, but that's a conversation we'll have with each one once this library is available.

Reply:
There are conceivably wallets that may be able to silently populate the accounts array, for example by making a network request or using a third party cookie with interaction, but that shouldn't make that request until the `connect` method is called. I don't know that we should eliminate being able to use the `silent` flag, it's still an intent hint that may be useful to the wallet depending on design and implementation which is unknown to the app.

Reply:
Super excellent point. I think there are two ways they should do that that don't involve connecting, semantically.

```ts
let firstTimeAccountLookup = true;
// A wallet adapter that has nothing to talk to, but might store the session from connection.
const MobileWalletAdapter: Wallet = {
    /* ... */
    get accounts() {
        if (firstTimeAccountLookup) {
            const savedAccountsFromLastTime = window.localStorage.get('mwaSavedAccounts');
            if (savedAccountsFromLastTime) {
                this.resolvedAccounts = JSON.parse(savedAccountsFromLastTime);  // whatever
            }
        }
        firstTimeAccountLookup = false;
        /* do normal thing */
    }
};
```

Or…

```ts
// A wallet adapter that can consult with a 3rd party upon construction.
class TipLinkGoogleWalletAdapter extends Wallet {
    constructor() {
        goAskGoogleForAccountsOrWhatever()
            .then(accounts => {
                this.accounts = accounts.map(getWalletAccountFromGoogleResponse);
                this.emit('standard:events', { accounts: this.accounts });
            });
    }
}
```

So tl;dr, there are upteen ways that a wallet can populate the `accounts` array and broadcast the fact that has happened that don't involve the app calling `connect()`.

Reply:
Just read this part of your review again:

> …but that shouldn't make that request until the connect method is called.

Why shouldn't?

Also, if (shouldn't === shouldn't do that work until the app has demonstrated intent to _read_ the `accounts` array), then they could kick that work off in a getter.

```ts
class TipLinkGoogleWalletAdapter extends Wallet {
    #shouldGoGetAccountsFromGoogleOrWhatever = false;
    get accounts() {
        if (this.#shouldGoGetAccountsFromGoogleOrWhatever) {
            goAskGoogleForAccountsOrWhatever()
                .then(accounts => {
                    this.resolvedAccounts = accounts.map(getWalletAccountFromGoogleResponse);
                    this.emit('standard:events', { accounts: this.resolvedAccounts });
                });
        }
        #shouldGoGetAccountsFromGoogleOrWhatever = false;
        return this.resolvedAccounts;
    }
}
```

Reply:
Since we likely intend to pass other arguments to `connect` (`features`, `chains` at minimum), should we just allow passing the args and not pass `silent: false`, which is already the default behavior?

Reply:
I guess where I land on this issue is as follows:

* `silent: true` has no use cases. All of your prior account authorizations should just materialize on `accounts` if they exist, without the need to call `connect()` at all
* What an app is expressing by calling connect is: ‘I see the zero-or-more accounts in the `accounts` array, but what else ya got for me?’ ie. you call `connect()` to seek additional account authorizations

I'm totally fine with piping through args to the underlying connect feature, but I'd like to omit `silent`.

```ts
(args: Omit<Parameters<typeof connectFeature['connect']>[0], 'silent'>) => {
    connectFeature.connect(rest);
}
```

Reply:
I think that's fine, as long as we're confident that wallets will fix this. Because AFAIK this isn't going to be usable for any wallet and will block any app that expects autoconnect from using this. `connect({silent: true})` is AFAIK the only way to make this work currently.

In general I don't think most Solana wallets have a good implementation of `silent`, and I'm not sure they particularly can with their single selected account model. I  also agree that using `accounts` properly means `silent: true` doesn't do anything, that's what I found with Orbit. But it will stop apps using this new stuff until wallets fix their approach here.

Reply:
> `connect({silent: true})` is AFAIK the only way to make this work currently.

Right. ‘Auto-connect’ right now depends on apps mashing the `connect()` method until something happens. This is an anti-pattern and rather what we want is for wallets to populate the `accounts` array immediately upon registration. That's the Wallet Standard's version of ‘auto-connect.’